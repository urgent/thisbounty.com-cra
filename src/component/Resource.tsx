import React, { ReactChildren } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

import {
  faUsersCog,
  faServer,
  faFileCode,
  faBook
} from '@fortawesome/free-solid-svg-icons'

type Resource = 'programmers' | 'servers' | 'scripts' | 'libraries'

function lookup (icon: Resource) {
  switch (icon) {
    case 'programmers':
      return faUsersCog
    case 'servers':
      return faServer
    case 'scripts':
      return faFileCode
    case 'libraries':
      return faBook
  }
}

export function Resource ({
  icon,
  count,
  children
}: {
  icon: Resource
  count: number
  children?: ReactChildren
}) {
  return (
    <div>
      <FontAwesomeIcon icon={lookup(icon)}></FontAwesomeIcon> x {String(count)}
      {children}
    </div>
  )
}
