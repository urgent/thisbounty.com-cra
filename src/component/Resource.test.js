import renderer from 'react-test-renderer';
import React from 'react'
import { Resource } from './Resource'

test('Resource renders correctly', () => {
    const empty = renderer
        .create(<Resource fill={false} color="#F00" />)
        .toJSON();
    const filled = renderer
        .create(<Resource fill={true} color="#F00" />)
        .toJSON();
    expect(empty).toMatchSnapshot();
    expect(filled).toMatchSnapshot();
});