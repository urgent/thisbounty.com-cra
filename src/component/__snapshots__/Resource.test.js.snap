// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resource renders correctly 1`] = `
<div>
   x 
  undefined
</div>
`;

exports[`Resource renders correctly 2`] = `
<div>
   x 
  undefined
</div>
`;
